kaleabendrias/chat_app-Nestjs-
# feat: messaging, join room

## Pull Request Description

Features Added:
- Chat Room Management:
  - Create Room: Implemented functionality to create a new chat room with validation and user authentication.
  - join Room: Users can now join a chat room and receive the chat history.
- User Authentication:
  - Login/Signup: Added user login and signup endpoints with validation and error handling.

** Enhancements:
- Improved real-time communication with WebSocket support for chat functionalities.
- Added authentication guards for secure room creation and messaging.
- Integrated Swagger documentation for API clarity.

Testing Instructions:

1. Join Room:
   - Connect to the WebSocket server.
   - Emit a `joinRoom` event with a valid `chatRoomId` and verify the client receives the chat history.

2. Create Room:
   - Emit a `createRoom` event with a valid `CreateRoomDto` payload.
   - Verify that the room is created and returned successfully, or check for validation errors.

3. Login/Signup:
   - Signup: Send a POST request to `/auth/register` with valid `CreateUserDto` data and verify user registration.
   - Login: Send a POST request to `/auth/login` with valid `LoginDto` data and verify successful authentication and JWT issuance.

Notes:
- Ensure that WebSocket connections are authenticated before accessing chat functionalities.
- Review Swagger documentation for detailed API specifications.
